Sources/Filters/Core/CleanPolyData/index.js

```diff
@@ -0,0 +1,488 @@
+import macro from 'vtk.js/Sources/macros';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import vtkMergePoints from 'vtk.js/Sources/Common/DataModel/MergePoints';
+import vtkPointLocator from 'vtk.js/Sources/Common/DataModel/PointLocator';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData/index';
+import vtkBoundingBox from 'vtk.js/Sources/Common/DataModel/BoundingBox';
+import { DesiredOutputPrecision } from 'vtk.js/Sources/Common/DataModel/DataSetAttributes/Constants';
+import { VtkDataTypes } from 'vtk.js/Sources/Common/Core/DataArray/Constants';
+
+// ----------------------------------------------------------------------------
+// vtkCleanPolyData methods
+// ----------------------------------------------------------------------------
+
+function vtkCleanPolyData(publicAPI, model) {
+  // Set our classname
+  model.classHierarchy.push('vtkCleanPolyData');
+
+  const tempX = [];
+
+  // Point processing
+  function processPoint(
+    ptId,
+    inPts,
+    newPts,
+    inputPD,
+    outputPD,
+    pointMap,
+    numUsedPts
+  ) {
+    const newX = [0, 0, 0];
+
+    inPts.getPoint(ptId, tempX);
+    publicAPI.operateOnPoint(tempX, newX);
+
+    if (!model.pointMerging) {
+      if (pointMap[ptId] === -1) {
+        pointMap[ptId] = numUsedPts.value++;
+        newPts.setPoint(pointMap[ptId], newX);
+        outputPD.passData(inputPD, ptId, pointMap[ptId]);
+      }
+      return pointMap[ptId];
+    }
+    const newPtId = model.locator.insertUniquePoint(newX).id;
+    if (!model.copiedPoints.has(newPtId)) {
+      model.copiedPoints.add(newPtId);
+      outputPD.passData(inputPD, ptId, newPtId);
+    }
+    return newPtId;
+  }
+
+  publicAPI.operateOnPoint = (inPt, outPt) => {
+    outPt[0] = inPt[0];
+    outPt[1] = inPt[1];
+    outPt[2] = inPt[2];
+  };
+
+  publicAPI.operateOnBounds = (inBounds, outBounds) => {
+    vtkBoundingBox.setBounds(outBounds, inBounds);
+  };
+
+  publicAPI.createDefaultLocator = (input) => {
+    let tol;
+    if (model.toleranceIsAbsolute) {
+      tol = model.absoluteTolerance;
+    } else if (input) {
+      tol = model.tolerance * input.getLength();
+    } else {
+      tol = model.tolerance;
+    }
+
+    if (!model.locator) {
```

Comment:
`model.locator` should probably be a "protected" member variable, it must be prefixed by '_'.
(or moved outside of model if no subclass could benefit from having access to it).